Include/internal/pycore_interp.h

```diff
@@ -230,6 +235,27 @@ PyAPI_FUNC(int) _PyInterpreterState_IDInitref(PyInterpreterState *);
 PyAPI_FUNC(int) _PyInterpreterState_IDIncref(PyInterpreterState *);
 PyAPI_FUNC(void) _PyInterpreterState_IDDecref(PyInterpreterState *);
 
+#define FREELIST_QUANTUM (2*sizeof(void*))
+#define SIZE_TO_FREELIST_INDEX(size) ((size-4)/FREELIST_QUANTUM)
```

Comment:
Why this formula?

You want to allocate all sizes from `(n-1)*FREELIST_QUANTUM + 1` to `n*FREELIST_QUANTUM` (inclusive) from the same freelist.
I would expect the formula to be `(size + FREELIST_QUANTUM -1)/FREELIST_QUANTUM`, or `(size-1)/FREELIST_QUANTUM`.

And because C division rounds towards 0, not -infinity, you want to use `>>LOG_BASE_2_OF_FREELIST_QUANTUM`, not `/FREELIST_QUANTUM`.

Reply:
I haven't changed this yet. I'm trying to get it to work with just ints.

Reply:
There already are some macros in pycore_obmalloc.h that seem to do something like this:

```
/* 
 * Alignment of addresses returned to the user. 8-bytes alignment works
 * on most current architectures (with 32-bit or 64-bit address buses).
 * The alignment value is also used for grouping small requests in size
 * classes spaced ALIGNMENT bytes apart.
 *
 * You shouldn't change this unless you know what you are doing.
 */

#if SIZEOF_VOID_P > 4
#define ALIGNMENT              16               /* must be 2^N */
#define ALIGNMENT_SHIFT         4
#else
#define ALIGNMENT               8               /* must be 2^N */
#define ALIGNMENT_SHIFT         3
#endif
   
/* Return the number of bytes in size class I, as a uint. */
#define INDEX2SIZE(I) (((pymem_uint)(I) + 1) << ALIGNMENT_SHIFT)
```